lib/utils.bash

```diff
@@ -736,7 +736,8 @@ select_version() {
     version_and_path=$(find_versions "$plugin_name" "$search_path")
     IFS='|' read -r version_string _path <<<"$version_and_path"
     IFS=' ' read -r -a usable_plugin_versions <<<"$version_string"
-    for plugin_version in "${usable_plugin_versions[@]}"; do
+    for version_spec in "${usable_plugin_versions[@]}"; do
+      plugin_version="$(resolve_version_spec "$version_spec")"
```

Comment:
This is the key change to enable the desired feature of this PR - now _resolving_ each version spec in the `.tool-versions` file to a precise version number, rather than it having to _already_ be a precise version number.